biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotations.java

```diff
@@ -14,11 +15,34 @@
  */
 public class DSAnnotations implements AnalyzerPlugin {
 
+	enum Flag {
+		inherit, felixExtensions, extender
+	};
+
 	public boolean analyzeJar(Analyzer analyzer) throws Exception {
 		Parameters header = OSGiHeader.parseHeader(analyzer.getProperty(Constants.DSANNOTATIONS));
 		if (header.size() == 0)
 			return false;
 
+		Parameters flagHeader = OSGiHeader.parseHeader(analyzer.getProperty(Constants.DSANNOTATIONS + "-flags"));
+		Set<Flag> flagSet = new HashSet<Flag>();
+		for (String s : flagHeader.keySet()) {
+			try {
+				flagSet.add(Enum.valueOf(Flag.class, s));
+			}
+			catch (IllegalArgumentException e) {
+				analyzer.error(
+						"Unrecognized -dsannotations-flags value %s, expected values are %s", s,
+						EnumSet.allOf(Flag.class));
+			}
+		}
+		if (Processor.isTrue(analyzer.getProperty("-dsannotations-inherit")))
```

Comment:
You need to create constants for -dsannotations-* instructions in aQute.bnd.osgi.Constants like DSANNOTATIONS for these instructions. Then these constants need to be mentioned in aQute.bnd.osgi.Constants.options and aQute.bnd.osgi.Constants.BUNDLE_SPECIFIC_HEADERS.